components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -0,0 +1,667 @@
+ */
+- (CGFloat)transitionPercentageForContentOffset:(CGPoint)contentOffset
+                                         offset:(CGFloat)offset
+                                       distance:(CGFloat)distance;
+
+/**
+ Checks the given target content offset to ensure the target offset will not cause the drawer to
+ end up in the middle of the header animation when the dragging ends. When needed, returns an
+ updated vertical target content offset that ensures the header animation is in a defined state.
+ Otherwise, returns NSNotFound.
+ */
+- (CGFloat)midAnimationScrollToPositionForOffset:(CGPoint)targetContentOffset;
+
+@end
+
+@interface MDCBottomDrawerContainerViewController (LayoutValues)
+
+// The presenting view's bounds.
+@property(nonatomic, readonly) CGRect presentingViewBounds;
+
+// Whether the content reaches to fullscreen.
+@property(nonatomic, readonly) BOOL contentReachesFullscreen;
+
+// Whether the content height exceeds the visible height when it's first displayed.
+@property(nonatomic, readonly) BOOL contentScrollsToReveal;
+
+// The top header height when the drawer is displayed in fullscreen.
+@property(nonatomic, readonly) CGFloat topHeaderHeight;
+
+// The content header height when the drawer is first displayed.
+@property(nonatomic, readonly) CGFloat contentHeaderHeight;
+
+// The vertical content offset where the transition animation completes.
+@property(nonatomic, readonly) CGFloat transitionCompleteContentOffset;
+
+// The headers animation distance.
+@property(nonatomic, readonly) CGFloat headerAnimationDistance;
+
+// The distance to top threshold for adding extra content height.
+@property(nonatomic, readonly) CGFloat addedContentHeightThreshold;
+
+@end
+
+@interface MDCBottomDrawerContainerViewController () <UIScrollViewDelegate>
+
+// Whether the scroll view is observed via KVO.
+@property(nonatomic) BOOL scrollViewObserved;
+
+// The scroll view is currently being dragged towards bottom.
+@property(nonatomic) BOOL scrollViewIsDraggedToBottom;
+
+// The scroll view has started its current drag from fullscreen.
+@property(nonatomic) BOOL scrollViewBeganDraggingFromFullscreen;
+
+// Whether the drawer is currently shown in fullscreen.
+@property(nonatomic) BOOL currentlyFullscreen;
+
+// Views:
+
+// The main scroll view.
+@property(nonatomic, readonly) UIScrollView *scrollView;
+
+// View that functions as the superview of |scrollView|. Used to clip the top of the scroll view.
+@property(nonatomic, readonly) UIView *scrollViewClippingView;
+
+// The top header bottom shadow layer.
+@property(nonatomic) MDCShadowLayer *headerShadowLayer;
+
+@end
+
+@implementation MDCBottomDrawerContainerViewController {
+  UIScrollView *_scrollView;
+  UIView *_scrollViewClippingView;
+  CGFloat _contentHeaderTopInset;
+  CGFloat _contentHeightSurplus;
+  CGFloat _addedContentHeight;
+}
+
+- (instancetype)initWithOriginalPresentingViewController:
+                    (UIViewController *)originalPresentingViewController
```

Comment:
If you cannot fit this on the line above, it should be indented +4 from the beginning of the line.
trackingScrollView below is fine as it is aligned on the colons.

Reply:
done